test/async-test.js

```diff
@@ -0,0 +1,62 @@
+'use strict'
+
+const t = require('tap')
+const test = t.test
+
+const fastify = require('fastify')()
+
+fastify
+  .decorate('verifyJWTandLevel', function (request, reply) {
+    return new Promise(function (resolve, reject) {
+      if (request.body && request.body.failure) {
+        reply.code(401).send({ 'error': 'Unauthorized' })
+        reject(new Error())
```

Comment:
I think here we shouldn't use the reply.send to send the error, but only reject with an error `reject(new Error('Password not valid')` and check that the output will be a status code 401

Reply:
done